src/fitnesse/testsystems/slim/tables/ScriptTable.java

```diff
@@ -297,15 +298,19 @@ protected boolean invokesSequentialArgumentProcessing(String cellContents) {
   public static class RowHelper {
     private static final String SEQUENTIAL_ARGUMENT_PROCESSING_SUFFIX = ";";
 
-    public static String getScenarioNameFromAlternatingCells(Table table, int endingCol, int row) {
+    public static String getScenarioNameFromAlternatingCells(Table table, int endingCol, int row) throws SyntaxError {
       String actionName = getActionNameStartingAt(table, 0, endingCol, row);
       String simpleName = actionName.replace(SEQUENTIAL_ARGUMENT_PROCESSING_SUFFIX, "");
       return Disgracer.disgraceClassName(simpleName);
     }
 
-    public static String getActionNameStartingAt(Table table, int startingCol, int endingCol, int row) {
+    public static String getActionNameStartingAt(Table table, int startingCol, int endingCol, int row) throws SyntaxError {
       StringBuilder actionName = new StringBuilder();
-      actionName.append(table.getCellContents(startingCol, row));
+      try {
+        actionName.append(table.getCellContents(startingCol, row));
+      } catch (IndexOutOfBoundsException e) {
+        throw new SyntaxError("badly formatted table: Expected an action name to be found at (" + row + "," + startingCol + ")");
```

Comment:
I believe the message could be a bit clearer. The [script table documentation](http://fitnesse.org/FitNesse.UserGuide.WritingAcceptanceTests.SliM.ScriptTable) does not mention 'action name' it uses the term 'function call'. 
Furthermore the 'catch block' of the `SyntaxException` already adds 'Bad table!' so that does not have to be repeated in the message.

What would you think of the following message: 
`"Too few columns in row " + (row + 1) + ". Expected a  function in column " + (startingCol + 1) + "."`

Reply:
I think it's better :)
Updating the PR now

Reply:
BTW, do you know where those error messages rendering happen in the code? I think it would make sense to add a test there, perhaps?

Reply:
`fitnesse.testsystems.slim.HtmlSlimTestSystem` line 57.

I don't believe adding a test there is needed, as it is very generic handling of SyntaxErrors coming from processing any Slim table. So having a unit test at ScriptTable level seems sufficient to me.